### DISCUSSION TITLE
Prusa XL with Bed Visualizer?

### DISCUSSION THREAD

**Discussion by technomadic42** [jneilliii/OctoPrint-BedLevelVisualizer#670] [Q&A] [answered]
Hey everyone.  So I've got My Prusa XL set up through octoprint and I've had pretty good success with it so far.

Where things seem to get a bit difficult is with the bed visualizer plugin.  It installs fine, runs fine and "updates the mesh", but in reality all it does is clean the nozzle by tapping it three times at the front edge of the plate and then returning a 0.0 perfectly flat result.

Am I missing something in the gcode for this plugin, or is it simply not going to work with this printer?  I know it works with the mk4, and had hoped that it would be somewhat similar with the XL.

Thanks for any insight anyone can offer to me on this.

**Comment by jneilliii** [comment]
You just have to figure out the right command to use for either running and reporting the mesh or just reporting the mesh.

**Reply by technomadic42** [reply]
Thanks,  I'll give it another try.  Maybe I missed something copying it in there.  Hoping that's the case and not that it doesn't work with the 6.1.2 firmware.

**Reply by technomadic42** [reply]
It works, and is COMICALLY off level, but prints all the same, and I suppose since there's no real way to update the Z offset or levelling on the XL, that's all mostly for curiosity.

Thanks for the prompt response!

**Comment by jneilliii** [comment, marked as answer]
https://github.com/jneilliii/OctoPrint-BedLevelVisualizer/blob/master/wiki/gcode-examples.md#prusa-xl5t-512-firmware
